sunpycube/cube/NDCube.py

```diff
@@ -132,6 +133,17 @@ def __getitem__(self, item):
         return result
 
 
+class NDCubeOrdered(NDCube):
+    """docstring for NDCubeOrdered"""
+
+    def __init__(self, data, uncertainty=None, mask=None, wcs=None, meta=None, unit=None, copy=False, missing_axis=None, **kwargs):
+        axtypes = list(wcs.wcs.ctype)
+        wcs_order = np.array(sunpycube.cube.cube_utils.select_order(axtypes))[::-1]
+        result_wcs = sunpycube.wcs_util.reindex_wcs(wcs, wcs_order)
```

Comment:
This has to alter both the shape of the array and the wcs. You can take out code from orient() in cube_utils that does this.  I think you just have to add the following:
```
array_order = select_order(axtypes)
result_data = array.transpose(array_order)
```
and then use ```result_data``` instead of ```data``` when calling the ```NDCube.__init__```.